packages/workflows/src/definition/product/prepare-create-inventory-items.ts

```diff
@@ -0,0 +1,46 @@
+import { ProductTypes } from "@medusajs/types"
+import { WorkflowArguments } from "../../helper"
+
+type AssociationTaggedVariant = ProductTypes.ProductVariantDTO & {
+  _associationTag?: string
+}
+
+type ObjectWithVariant = { variants: ProductTypes.ProductVariantDTO[] }
+
+export async function prepareCreateInventoryItems({
+  data,
+}: WorkflowArguments<{
+  products: ObjectWithVariant[]
+}>) {
+  const taggedVariants = data.products.reduce<AssociationTaggedVariant[]>(
+    (acc, product: ObjectWithVariant) => {
+      const cleanVariants = product.variants.reduce<AssociationTaggedVariant[]>(
+        (acc, variant: AssociationTaggedVariant) => {
+          if (!variant.manage_inventory) {
+            return acc
+          }
+
+          variant._associationTag = variant.id
+
+          acc.push(variant)
+          return acc
+        },
+        []
+      )
+      return acc.concat(cleanVariants)
+    },
+    []
+  )
+
```

Comment:
```suggestion
 const taggedVariants = data.products
  .map(product => {
    return product.variants.filter(variant => {
      variant._associationTag = variant.id
      return variant.manage_inventory
    })
  })
  .flat()
```

Reply:
This is gonna get changed in one of the other PRs any way so let's see if we can simplify there instead :)